src/ripple/shamap/impl/SHAMapDelta.cpp

```diff
@@ -112,7 +112,9 @@ bool SHAMap::walkBranch (SHAMapTreeNode* node,
     return true;
 }
 
-bool SHAMap::compare (SHAMap::ref otherMap, Delta& differences, int maxCount)
+bool
+SHAMap::compare (std::shared_ptr<SHAMap> const& otherMap,
+                 Delta& differences, int maxCount)
```

Comment:
A real non-const function being called or just too many functions to audit for constness?